Cabal-syntax/src/Distribution/Types/Flag.hs

```diff
@@ -111,7 +111,7 @@ instance Parsec FlagName where
     -- cabal check will do that.
     parsec = mkFlagName . lowercase <$> parsec'
       where
-        parsec' = (:) <$> lead <*> rest
+        parsec' = (<* P.skipSpaces) $ (:) <$> lead <*> rest
```

Comment:
IMHO, wrong place. This is better fixed in `FlagAssignment` parser (`Parsec` instance).

(We don't skip spaces after package names, etc. etc)

Reply:
fair enough. this seems to trip some tests too. i'll take another look.